modules/llama_cpp_plugin/src/compiled_model.cpp

```diff
@@ -0,0 +1,116 @@
+#include "compiled_model.hpp"
+
+#include <fstream>
+#include <memory>
+#include <openvino/op/constant.hpp>
+#include <openvino/opsets/opset13.hpp>
+#include <openvino/runtime/properties.hpp>
+#include <openvino/util/log.hpp>
+
+#include "infer_request.hpp"
+#include "plugin.hpp"
+
+namespace ov {
+namespace llama_cpp_plugin {
+
+LlamaCppModel::LlamaCppModel(const std::shared_ptr<ov::Model>& model,
+                             const std::shared_ptr<const ov::IPlugin>& plugin,
+                             const ov::SoPtr<ov::IRemoteContext>& context,
+                             const std::shared_ptr<ov::threading::ITaskExecutor>& task_executor)
+    : ICompiledModel(model, plugin, context, task_executor) {
+    OPENVINO_THROW_NOT_IMPLEMENTED("Currently only direct GGUF file loading is supported for the LLAMA_CPP* plugins");
+}
+
+LlamaCppModel::LlamaCppModel(const std::shared_ptr<ov::Model>& ov_model,
+                             std::istream& input_stream,
+                             const std::shared_ptr<const IPlugin>& plugin)
+    : ICompiledModel(ov_model, plugin) {
+    OPENVINO_THROW_NOT_IMPLEMENTED("Currently only direct GGUF file loading is supported for the LLAMA_CPP* plugins");
+}
+
+LlamaCppModel::~LlamaCppModel() {
+    llama_free(m_llama_ctx);
+    llama_free_model(m_llama_model_ptr);
+    llama_backend_free();
+    delete num_tokens_processed_ptr;
+}
+
+LlamaCppModel::LlamaCppModel(const std::string& gguf_fname, const std::shared_ptr<const IPlugin>& plugin)
+    : ICompiledModel(nullptr, plugin),
+      m_gguf_fname(gguf_fname) {
+    num_tokens_processed_ptr = new size_t;  // TODO (vshampor): hack, remove
+    *num_tokens_processed_ptr = 0;
+    OPENVINO_DEBUG << "llama_cpp_plugin: loading llama model directly from GGUF... " << std::endl;
+    llama_model_params mparams = llama_model_default_params();
+    mparams.n_gpu_layers = 99;
+    m_llama_model_ptr = llama_load_model_from_file(gguf_fname.c_str(), mparams);
+    llama_context_params cparams = llama_context_default_params();
+    m_llama_ctx = llama_new_context_with_model(m_llama_model_ptr, cparams);
+    OPENVINO_DEBUG << "llama_cpp_plugin: llama model loaded successfully from GGUF..." << std::endl;
+
+    auto input_ids = std::make_shared<ov::opset13::Parameter>(ov::element::Type_t::i64, ov::PartialShape({-1, -1}));
+    auto fake_convert = std::make_shared<ov::opset13::Convert>(input_ids->output(0), ov::element::Type_t::f32);
+    auto logits = std::make_shared<ov::opset13::Result>(fake_convert->output(0));
+
+    ov::ParameterVector inputs{input_ids};
+
+    std::vector<std::pair<std::string, ov::element::Type_t>> unused_names_in_order = {
+        {"attention_mask", ov::element::Type_t::i64},
+        {"position_ids", ov::element::Type_t::i64},
+        {"beam_idx", ov::element::Type_t::i32}};
+    for (const auto& descr : unused_names_in_order) {
+        auto unused_inp = std::make_shared<ov::opset13::Parameter>(descr.second, ov::PartialShape({-1, -1}));
```

Comment:
looks like `beam_idx` should be of `{-1}` shape?

Reply:
Will do, but somehow the test worked even with the current version

Reply:
because you have not set this parameter? :-)

Reply:
Done